chain/client/src/stateless_validation/partial_witness_tracker.rs

```diff
@@ -91,13 +96,21 @@ impl CacheEntry {
         if self.data_parts_present < self.data_parts_required {
             return None;
         }
-        self.is_decoded = true;
         match rs_decode(&rs, &mut self.parts, encoded_length) {
-            Ok(encoded_chunk_state_witness) => Some(encoded_chunk_state_witness),
+            Ok(encoded_chunk_state_witness) => {
+                self.is_decoded = true;
+                Some(encoded_chunk_state_witness)
```

Comment:
shouldn't we also verify that fields in each of the witness parts (such as `epoch_id`,  `shard_id`, etc.) correspond to `encoded_chunk_state_witness`?